okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpClientStream.java

```diff
@@ -228,6 +229,14 @@ protected void returnProcessedBytes(int processedBytes) {
 
   @Override
   protected void sendCancel(Status reason) {
+    synchronized (lock) {
+      if (pendingData != null) {
+        // stream is pending.
+        transport.removePendingStream(this);
+        transportReportStatus(reason, true, new Metadata.Trailers());
```

Comment:
Hmm... should we do `requestHeaders = null;` and `buffer.clear()` all the entries of `pendingData`? Otherwise those objects may be saved for a while.